src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/ConfigCommands/ConfigRunners.cs

```diff
@@ -54,5 +96,108 @@ public static ISettings GetSettingsFromDirectory(string directory)
                 configFileName: null,
                 machineWideSettings: new XPlatMachineWideSetting());
         }
+
+        /// <summary>
+        /// Returns a string holding the value of the key in the config section
+        /// of the settings. If showPath is true, this will also return the path
+        /// to the configuration file where the key is located.
+        /// </summary>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static string GetValueForConfigKey(ISettings settings, string key, bool showPath)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            var sectionElement = settings.GetSection(ConfigurationConstants.Config);
+            var item = sectionElement?.GetFirstItemWithAttribute<AddItem>(ConfigurationConstants.KeyAttribute, key);
+
+            if (item == null)
+            {
+                return null;
+            }
+
+            if (showPath)
+            {
+                return item.Value + "\tfile: " + item.GetConfigFilePath();
+            }
+            return item.Value;
+        }
+
+        /// <summary>
+        /// Logs each section of the configuration settings that will be applied, grouped by file path.
+        /// </summary>
+        public static void LogSectionsWithPaths(Dictionary<string, VirtualSettingSection> sections, ILogger logger)
+        {
+            foreach (var section in sections)
```

Comment:
This will result in a null reference exception (NRE) when the `NullSettings` object is used.